File: src/commands/add.ts
import { readConfigFile, writeConfigFile } from "../executers/config";
import { createEndpointFiles } from "../executers/endpoint/createEndpointFiles";
import { getRepositoryPath } from "../executers/repository/getRepositoryPath";
import { getRepositoryAlias } from "../executers/repository/getRepositoryAlias";

export const add = async (
  repositoryName: string,
  workspaces?: string[],
  branch?: string,
  exclduePeriods?: string[],
) => {
  const config = await readConfigFile();
  const alias = getRepositoryAlias(repositoryName);
  const repositoryPath = getRepositoryPath(repositoryName);

  const service = await createEndpointFiles(
    alias,
    {
      version: undefined,
      repository: repositoryPath,
      workspaces,
      branch,
      exclude_periods: exclduePeriods,
    },
    config.environment_identifier,
    config.output,
  );

  config.push(alias, service);

  await writeConfigFile(config);
};

File: src/commands/add.rs
use crate::executers::config::*;
use crate::executers::endpoint::*;
use crate::executers::repository::*;
use crate::model::config::Service;
use anyhow::{Ok, Result};

pub fn run(
    repository_name: String,
    workspaces: Option<Vec<String>>,
    branch: Option<String>,
    exclude_periods: Option<Vec<String>>,
) -> Result<()> {
    let mut config = read_config_file()?;

    let alias = get_repository_alias(&repository_name)?;
    let repository_path = get_repository_path(&repository_name)?;

    let service = create_endpoint_files(
        alias.clone(),
        Service {
            version: None,
            repository: repository_path,
            workspaces,
            branch,
            exclude_periods,
            roots: None,
        },
        config.environment_identifier.clone(),
        config.output.clone(),
    )?;

    config.push(alias, service);

    write_config_file(config)?;
    Ok(())
}
